gmso/core/topology.py

```diff
@@ -1417,6 +1506,220 @@ def __str__(self):
+                        )
+                        for connection in getattr(self, parameter)
+                    )
+                    df["z Atom" + str(site_index) + " (nm)"] = list(
+                        _return_float_for_unyt(
+                            getattr(
+                                connection.connection_members[site_index],
+                                "position",
+                            )[2],
+                            unyts_bool,
+                        )
+                        for connection in getattr(self, parameter)
+                    )
+                elif attr == "charge" or attr == "charges":
+                    df["charge Atom" + str(site_index) + " (e)"] = list(
+                        getattr(
+                            connection.connection_members[site_index],
+                            "charge",
+                        ).to_value()
+                        / u.electron_charge.to_value()
+                        for connection in getattr(self, parameter)
+                    )
+                else:
+                    try:
+                        df[attr + " Atom" + str(site_index)] = list(
+                            _return_float_for_unyt(
+                                getattr(
+                                    connection.connection_members[site_index],
+                                    attr,
+                                ),
+                                unyts_bool,
+                            )
+                            for connection in getattr(self, parameter)
+                        )
+                    except AttributeError:
+                        raise AttributeError(
+                            "The attribute {} is not in this gmso object".format(
+                                attr
+                            )
+                        )
+        else:
+            raise AttributeError(
+                "{} is not yet supported for adding labels to a dataframe. \
+                                 Please use  one of 'sites', 'bonds', 'angles', 'dihedrals', or \
+                                 'impropers'".format(
+                    str(parameter)
+                )
+            )
+        return df
+
+    def _parse_parameter_expression(self, df, parameter, unyts_bool):
+        """Take a given topology attribute and return the parameters associated with it"""
+        for i, param in enumerate(
+            getattr(
+                getattr(self, parameter)[0], parameter[:-1] + "_type"
+            ).parameters
+        ):
+            df[
+                "Parameter "
+                + str(i)
+                + " ("
+                + str(param)
+                + "): "
+                + str(
+                    getattr(
+                        getattr(self, parameter)[0], parameter[:-1] + "_type"
+                    )
+                    .parameters[param]
+                    .units
+                )
```

Comment:
I think this can be shortened for readability, but maybe task for another day.

Reply:
Fixed in d21d2d6b771209a4d2e1b4770267271707f89b17